ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -882,20 +882,24 @@ public CompletableFuture<RaftClientReply> submitClientRequestAsync(
     try {
       assertLifeCycleState(LifeCycle.States.RUNNING);
     } catch (ServerNotReadyException e) {
-      final RaftClientReply reply = newExceptionReply(request, e);
       requestRef.release();
+      final RaftClientReply reply = newExceptionReply(request, e);
```

Comment:
The original order (`newExceptionReply` -> `release`) seems better since `request` is used in `newExceptionReply`.

Reply:
Right, I made that change my mistake.